File: 78-subsets/78-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        size = len(nums)
        result = []
        for i in range(size):
            if i == 0:
                result = [[], [nums[i]]]
            else:
                result += [[nums[i]] + item for item in result]
            
        return result

File: 78-subsets/78-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        Arrays.sort(nums);
       
        
       List<List<Integer>> res = new ArrayList<>();
        res.add(new ArrayList<>());
        
          int end = 0;
        for(int i = 0; i < nums.length; i++) {
            int e = nums[i];
            int start = i > 0 && nums[i] == nums[i - 1] ? end : 0;
            end = res.size();
            for(int j = start; j < end; j++) {
                List<Integer> sub = new ArrayList<>(res.get(j));
                sub.add(e);
                res.add(sub);
            }
        }
        return res;
        
        
        
    }
}